docs/user-guide/best-practices/udf-best-practices.mdx

```diff
@@ -0,0 +1,311 @@
+- Files are read multiple times and >100MB
+- Files that are slow or require some processing to open (`.zip` for example)
+
+### Don't start from scratch: [UDF Catalog](/workbench/udf-catalog/)
+
+Just like using libraries in Python to leverage existing tools, you don't need to start from scratch in Fused. We have a [Catalog of existing UDFs](/workbench/udf-catalog/) built & maintained by us and the community
+
+You can find a host of different UDFs that can serve as a starting point or as inspiration to create your own UDFs:
+- Open datasets from common open repository [like STAC catalogs](https://www.fused.io/workbench/catalog/Sentinel_Tile_Example_2-ddb4c495-40e3-48ba-849c-256487f8a9cb)
+- Run [on the fly ML prediction](https://www.fused.io/workbench/catalog/Dl4eo_Airplane_Detection_Global-91f81c0c-08f6-4a0c-82f9-bd2e0322b4e7) on satellite images
+- Compute a [spatial join](https://www.fused.io/workbench/catalog/Overture_Nsi-dd89972c-ce30-4544-ba0f-81fc09f5bbef) between 2 datasets
+
+![UDF Catalog](/img/udfcatalog_august.png)
+
+You can also contribute your own UDFs to the community!
+
+## Debugging UDFs 
+
+The reality of writing code is that stuff breaks, often and sometimes in mysterious ways. Here's some of our recommendations for how to debug your UDFs
+
+### Use `print()`
+
+UDFs return `stdout` either in [Workbench Code Editor](/workbench/udf-builder/code-editor/) or locally when running `fused.run(udf)` so the easiest way to get info about your UDFs is to use good old `print`:
+
+```python showLineNumbers
+@fused.udf
+def udf(n: int = 1):
+    print(f"{n=}")
+    return
+```
+
+Since Python 3.8 you can use [f-string debugging](https://docs.python.org/3/whatsnew/3.8.html#f-strings-support-for-self-documenting-expressions-and-debugging) which is what we recommend you use:
+```python showLineNumbers
+print(f"{my_fancy_variable=}")
+```
+
+This allows you to print many variables without getting lost with what is what
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs className="unique-tabs">
+  <TabItem value="workbench" label="Workbench" default>
+    ![UDF Catalog](/img/user-guide/best-practices/workbench_print_stdout.png)
+  </TabItem>
+  <TabItem value="notebook" label="notebook" default>
+    ![UDF Catalog](/img/user-guide/best-practices/notebook_print_stdout.png)
+  </TabItem>
+</Tabs>
+
+### Type all your inputs
+
+We strongly recommend you [type](https://docs.python.org/3/library/typing.html) all your inputs with the appropriate type:
+
+```python {3} showLineNumbers
+@fused.udf
+def udf(
+    bbox:fused.types.TileGDF=None, n:int=1
+):
+    ...
+    return
+```
+
+This has 2 effects:
+- It makes your code more readable to others
+- Fused only supports a [few types](/core-concepts/write/#supported-types) at the moment. Any non-typed or unsupported types will be handled as `str`
```

Comment:
```suggestion
- Fused only supports a [few types](/core-concepts/write/#supported-types) at the moment. Any non-typed or unsupported types will be passed as `str`
```

? to make it more clear that the UDF function will receive it as strings (when running remotely)